amgut/handlers/interactive.py

```diff
@@ -0,0 +1,195 @@
+            for otu in unseen:
+                otus[otu].append(0.0)
+
+        datasets = _build_taxa(otus)
+
+        meta_cats = ['age-baby', 'age-child', 'age-teen', 'age-20s', 'age-30s',
+                     'age-40s', 'age-50s', 'age-60s', 'age-70+',
+                     'bmi-Underweight', 'bmi-Normal', 'bmi-Overweight',
+                     'bmi-Obese', 'sex-male', 'sex-female', ]
+
+        with open(join(AMGUT_CONFIG.base_data_dir, 'emperor',
+                  'emperor.txt')) as f:
+            pcoa_data = f.readlines()
+        self.render("emperor.html", barcodes=bc_info,
+                    coords_ids=pcoa_data[0].strip(),
+                    coords=pcoa_data[1].strip(),
+                    pct_var=pcoa_data[2].strip(),
+                    md_headers=literal_eval(pcoa_data[3].strip()),
+                    metadata=pcoa_data[4].strip(),
+                    titles=titles, meta_cats=meta_cats, datasets=datasets)
+
+
+def _build_taxa(taxa_list):
+    def clean(taxa):
+            cleaned = taxa.split("__")[1]
+            if "[" in cleaned:
+                cleaned = "%s (Contested)" % cleaned.translate(None, '[]')
+            return cleaned
+
+    datasets = []
+    for otu, value in taxa_list.items():
+        taxonomy = [x.strip() for x in otu.split(';')]
+        # Find the highest classified level by looping backwards through
+        # the list and breaking once we find the first classified level
+        highest = ''
+        for tax in reversed(taxonomy):
+            if not tax.endswith('__'):
+                highest = 'Unclassified (%s)' % tax.replace('__', '. ')
+                break
+
+        datasets.append({
+            'full': otu,
+            'phylum': taxonomy[1].split("__")[1],
+            'class': clean(taxonomy[2]) if not
+            taxonomy[2].endswith('__') else highest,
+            'order': clean(taxonomy[3]) if not
+            taxonomy[3].endswith('__') else highest,
+            'family': clean(taxonomy[4]) if not
+            taxonomy[4].endswith('__') else highest,
+            'genus': clean(taxonomy[5]) if not
+            taxonomy[5].endswith('__') else highest,
+            'data': value
+        })
+    return datasets
+
+
+class TaxaHandler(BaseHandler):
+    @authenticated
+    def get(self):
+        user = ag_data.get_user_for_kit(self.current_user)
+        bc_info = ag_data.get_barcodes_by_user(user, results=True)
+        titles = [bc['sample_date'].strftime('%b %d, %Y') for bc in bc_info]
+        otus = {}
+        files = []
+        # Load in all possible OTUs that can be seen by loading files to memory
+        # Files are small (5kb) so this should be fine.
+        for barcode in bc_info:
+            with open(join(AMGUT_CONFIG.base_data_dir, 'taxa-summaries',
+                      '%s.txt' % barcode['barcode'])) as f:
+                f.readline()
+                files.append(f.readlines())
+            for line in files[-1]:
```

Comment:
Oh, I see now. The code is always just pulling the parsed data from the last file parsed. 

I think it would be helpful to break the parsing logic out. I'm personally finding it difficult to wrap my head around and I'm not entirely clear as to why the parsing needs to be as complex as it is